db.go

```diff
@@ -67,6 +81,18 @@ const emptyReputationTableSQL = `
 TRUNCATE TABLE reputation;
 `
 
+const createViolationReputationWeightsTableSQL = `
+CREATE TABLE IF NOT EXISTS violation_reputation_weights (
+violation_type varchar(128) PRIMARY KEY NOT NULL,
+reputation int NOT NULL CHECK (reputation >= 0 AND reputation <= 100),
```

Comment:
A reputation should be between 0 and 100, and processing a violation must not **increase** the reputation of an IP, only decrease it. (if IP has reputation of 20 and violation comes in which does a minus 40, the final reputation should be zero, not 60).